### PR TITLE
RPC: Remove `anchor_txid` from `AnchorInfo`

### LINKED ISSUE #654: Remove field `anchor_txid` from RPC message `AnchorInfo`.
The txid is included in field `anchor_outpoint`. We should therefore remove field `anchor_txid` from the `AnchorInfo` RPC message.

### PR DESCRIPTION
**Pull request by GeorgeTsagk**
Closes #654

### CODE REVIEW

File: itest/assertions.go [hunk 1]
```diff
@@ -69,9 +70,12 @@ func AssetAnchorCheck(txid, blockHash chainhash.Hash) AssetCheck {
 			return fmt.Errorf("asset is missing chain anchor field")
 		}
 
-		if a.ChainAnchor.AnchorTxid != txid.String() {
+		anchorTxid :=
+			strings.Split(a.ChainAnchor.AnchorOutpoint, ":")[0]
```
Comment (GeorgeTsagk) [inline itest/assertions.go:74]: Since the txid info was duplicate I now retrieve it by splitting the outpoint string

**Comment by GeorgeTsagk** [comment]
There was a flake on a batch mint assets test, kicked it and passed, just flagging

File: itest/assertions.go (hunk 1 above)
Comment (ffranr) [reply itest/assertions.go:74]: I think you should use `wire.NewOutPointFromString` here.

Comment (ffranr) [reply itest/assertions.go:74]: I think you should use `wire.NewOutPointFromString` here.

**Review by ffranr — CHANGES_REQUESTED** [review]

File: itest/assertions.go (hunk 1 above)
Comment (GeorgeTsagk) [reply itest/assertions.go:74]: Good call, is the hash reversed in the raw outpoint string?

File: itest/assertions.go [hunk 2]
```diff
@@ -69,9 +69,18 @@ func AssetAnchorCheck(txid, blockHash chainhash.Hash) AssetCheck {
 			return fmt.Errorf("asset is missing chain anchor field")
 		}
 
-		if a.ChainAnchor.AnchorTxid != txid.String() {
+		out, err :=
+			wire.NewOutPointFromString(a.ChainAnchor.AnchorOutpoint)
+
```
Comment (ffranr) [inline itest/assertions.go:74]: nit: this empty line is unnecessary

File: itest/assertions.go [hunk 3]
```diff
@@ -69,9 +69,18 @@ func AssetAnchorCheck(txid, blockHash chainhash.Hash) AssetCheck {
...
+		if err != nil {
+			return fmt.Errorf("unable to parse outpoint: %v", err)
+		}
+
+		anchorTxid := out.Hash.String()
+
+		if anchorTxid != txid.String() {
 			return fmt.Errorf("unexpected asset anchor TXID, got "+
-				"%v wanted %x", a.ChainAnchor.AnchorTxid,
+				"%v wanted %x", anchorTxid,
```
Comment (ffranr) [inline itest/assertions.go:83]: nit: the `txid[:]` can be on the same line as `"%v wanted %x", anchorTxid,`

File: itest/assets_test.go [hunk 4]
```diff
@@ -173,11 +174,15 @@ func transferAssetProofs(t *harnessTest, src, dst *tapdHarness,
...
-		anchorTxHash, err := chainhash.NewHashFromStr(
-			existingAsset.ChainAnchor.AnchorTxid,
+		out, err := wire.NewOutPointFromString(
+			existingAsset.ChainAnchor.AnchorOutpoint,
 		)
 		require.NoError(t.t, err)
 
+		anchorTxHash := out.Hash
+
+		require.NoError(t.t, err)
```
Comment (ffranr) [inline itest/assets_test.go:184]: This line is redundant. We're doing the same thing two statements above.

**Review by ffranr — CHANGES_REQUESTED** [review]
Looks good to me! Just two or so nits.

I'm a fan of this change because it reduces our test surface.

File: itest/assertions.go (hunk 1 above)
Comment (ffranr) [reply itest/assertions.go]: Yes, the outpoint string consists of the `txid` and the `vout` index. The `txid` in the outpoint string is a hex encoding of the byte reversed hash.

**Review by ffranr — APPROVED** [review]
Nice! :100:

**Review by jharveyb — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**